manifests/prepare/puppet_config.pp

```diff
@@ -26,7 +26,14 @@
       # Deprecated for global config
       'config_version', 'manifest', 'modulepath',
       # Settings that should be reset to defaults
-      'disable_warnings', 'vardir', 'rundir', 'libdir', 'confdir', 'ssldir', 'classfile'].each |$setting| {
+      'disable_warnings', 'vardir', 'rundir', 'libdir', 'confdir', 'ssldir', 'classfile']
+    # When upgrading to 1.4.x or later remove pluginsync
+    if versioncmp($::puppet_agent::package_version, '1.4.0') >= 0 {
```

Comment:
This only works if we move https://github.com/puppetlabs/puppetlabs-puppet_agent/blob/master/manifests/init.pp#L38-L42 to params.pp.